src/views/AddItem.jsx

```diff
@@ -1,3 +1,115 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useEffect, useState } from 'react';
+import { addItem } from '../api/firebase';
+
+const timeframeToDays = {
+	Soon: 7,
+	'Kind of Soon': 14,
+	'Not Soon': 30,
+};
+
+export function AddItem({ listId }) {
+	const [timeframe, setTimeframe] = useState('Soon');
+	const [itemName, setItemName] = useState('');
+	const [message, setMessage] = useState(null);
+
+	useEffect(() => {
+		const timer = setTimeout(() => {
+			setMessage(null);
+		}, 3000);
+
+		return () => clearTimeout(timer);
+	}, [message]);
+
+	const onTimeChange = (e) => setTimeframe(e.target.value);
+	const onItemChange = (e) => setItemName(e.target.value);
+
+	return (
+		<form
+			onSubmit={(event) => {
+				event.preventDefault();
+				// Make sure the user has entered an item name
+				if (!itemName) {
+					setMessage('Please enter an item name');
+					return;
+				}
+
+				const result = addItem(listId, {
+					itemName,
+					daysUntilNextPurchase: timeframeToDays[timeframe],
+				});
+				if (result) {
+					setMessage(`Added ${itemName} to your list.`);
+					setItemName('');
+					setTimeframe('Soon');
+				} else {
+					setMessage('Error adding item');
```

Comment:
Were you all able to trigger this error message? I've tried a bunch of bad inputs, and I couldn't get the error.
`result` here is also a promise, so the null check doesn't reall work if the promise was rejected by Firebase